node-definitions/juniper/vsrx/jnpr-vsrx.yaml

```diff
@@ -0,0 +1,100 @@
+#
+# Juniper vSRX 3.0 virtual firewall node definition
+# generated 2023-01-09
+# part of VIRL^2
+#
+id: jnpr-vsrx
+configuration:
+  generator:
+    driver: null
+  provisioning:
+    volume_name: disk
+    media_type: iso
+    files:
+      - name: juniper.conf
+        content: hostname insert_hostname_here
```

Comment:
Does this work?  I'm more familiar with JunOS router syntax.  Just curious if this does, in fact, set the hostname.

Reply:
It doesn't work. vSRX will load factory default config after syntax checking. Boot logs here:

```
Creating initial configuration:  ...
/config/juniper.conf:1:(8) syntax error at 'hostname'
  [edit]
    'hostname insert_hostname_here'
      syntax error
mgd: warning: loading configuration from /config/juniper.conf
/config/juniper.conf:1:(8) syntax error at 'hostname'
  [edit]
    'hostname insert_hostname_here'
      syntax error
mgd: warning: activating factory configuration
UTM Daemon: <xnm:warning xmlns="http://xml.juniper.net/xnm/1.1/xnm" xmlns:xnm="http://xml.juniper.net/xnm/1.1/xnm">
UTM Daemon: <source-daemon>utmd</source-daemon>
UTM Daemon: <message>You must reboot the system for Web-Filtering's default mode to take effect.
UTM Daemon: </message>
UTM Daemon: </xnm:warning>
<output>Generating RSA2 key /etc/ssh/ssh_host_rsa_key</output>
```

I write this(`hostname insert_hostname_here`) to maintain consistency with vmx/vqfx's node definitions. Maybe empty file is better?

If you want to set the hostname at initial startup, you need to let `juniper.conf` contain full factory default config + `host-name insert_hostname_here`, which is more complex & no backward compatibility.

```
version 22.2R2-S1.5;
system {
    host-name insert_hostname_here;
    services {
        ssh;
        web-management {
            http {
                interface fxp0.0;
            }
            https {
                system-generated-certificate;
                interface fxp0.0;
            }
        }
    }
    syslog {
        file interactive-commands {
            interactive-commands any;
        }
        file messages {
            any any;
            authorization info;
        }                               
    }
    license {
        autoupdate {
            url https://ae1.juniper.net/junos/key_retrieval;
        }
    }
    ## Warning: missing mandatory statement(s): 'root-authentication'
}
security {
    screen {
        ids-option untrust-screen {
            icmp {
                ping-death;
            }
            ip {
                source-route-option;
                tear-drop;
            }
            tcp {
                syn-flood {
                    alarm-threshold 1024;
                    attack-threshold 200;
                    source-threshold 1024;
                    destination-threshold 2048;
                    queue-size 2000; ## Warning: 'queue-size' is deprecated
                    timeout 20;
                }
                land;
            }
        }
    }
    policies {
        from-zone trust to-zone trust {
            policy default-permit {
                match {
                    source-address any;
                    destination-address any;
                    application any;
                }
                then {
                    permit;
                }
            }
        }
        from-zone trust to-zone untrust {
            policy default-permit {     
                match {
                    source-address any;
                    destination-address any;
                    application any;
                }
                then {
                    permit;
                }
            }
        }
        pre-id-default-policy {
            then {
                log {
                    session-close;
                }
            }
        }
    }
    zones {
        security-zone trust {
            tcp-rst;
        }
        security-zone untrust {         
            screen untrust-screen;
        }
    }
}
interfaces {
    fxp0 {
        unit 0;
    }
}
```

Reply:
Please include that default config in the YAML file, then.  You can use the N9Kv YAML as an example of how to do this.  Your sample here is much more useful than an empty day 0 config.